server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardLoopIntegrationTest.java

```diff
@@ -144,16 +145,14 @@ void filterForwardShouldNotCreateLoopError() throws Exception {
 
         SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
             List<Mail> mails = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
-                .filter(Throwing.predicate(mail -> mail.getRecipients().contains(ALICE.asMailAddress())))
                 .collect(ImmutableList.toImmutableList());
 
-            softly.assertThat(mails.get(0).getRecipients()).containsOnly(ALICE.asMailAddress());
+            softly.assertThat(mails.get(0).getRecipients()).containsOnly(CEDRIC.asMailAddress());
         }));
     }
 
     @Test
     void regularForwardShouldNotCreateLoopError() throws Exception {
-        dataProbe.addMapping(MappingSource.fromUser(ALICE), Mapping.forward(SENDER.asString()));
```

Comment:
You could leave this one, and check you get a mail in cedric and sender. Covers an extra case IMO :)

Reply:
added a similar test case